src/authProvider/eosioAuthProvider.ts

```diff
@@ -0,0 +1,57 @@
+import { createLink, authenticate } from '3id-blockchain-utils'
+import type { LinkProof } from '3id-blockchain-utils'
+import { AccountID } from 'caip'
+
+import AbstractAuthProvider from './abstractAuthProvider'
+import { assert } from '../errors'
+import type { EOSIOProvider } from '../types'
+
+/**
+ *  AuthProvider which can be used for eosio providers that meet the universal authentication interface(UAL)
+ */
+class EOSIOAuthProvider extends AbstractAuthProvider {
+  accountId: string | null
+  provider: EOSIOProvider
+
+  constructor(eosioProvider: EOSIOProvider) {
+    assert.isDefined(eosioProvider, 'Missing EOSIO provider')
+    super('eosio')
+    this.provider = eosioProvider
+    this.accountId = null
+  }
+
+  async init(): Promise<void> {
```

Comment:
looks like this can be removed? both authenticate createLink call the same code paths, and not part of the standard interface

Reply:
I think its required, because the accountId property is accessed directly here:[https://github.com/ceramicstudio/3id-connect/blob/95a5513c8a1017e6e993c152b4650385a22a2442/src/threeIdConnect.ts#L174]( https://github.com/ceramicstudio/3id-connect/blob/95a5513c8a1017e6e993c152b4650385a22a2442/src/threeIdConnect.ts#L174)

Reply:
right ok, yeah we will likely change this a bit soon, we used to have init as part of standard interface, assuming some providers may need to make async call before they are ready, this is one example, but wont block this pr 👍